File: CPP/Algorithm/Repdigit_Numbers.cpp
// Complexity: O(n)
// Note: Repdigit Number is a number n which has all the digits in its representation in base b equal.

bool rd(int64_t n, int64_t b) {
	int64_t p = -1;
	while (n) {
		int64_t d = n % b; n /= b;
		if (p != -1 && d != p) return false;
		p = d;
	}
	return true;
}

File: Python/Algorithm/Repdigit_Numbers.py
# Complexity: O(n)
# Note: Repd nber is a nber n which has all the ds in its representation in base b equal.

def rd(n, b):
	p = -1
	while n:
		d = n % b; n //= b
		if p != -1 and d != p: return False
		p = d
	return True